fendermint/testing/contract-test/tests/run_upgrades.rs

```diff
@@ -0,0 +1,272 @@
+    upgrade_scheduler
+        .add(
+            Upgrade::new("mychain", 1, Some(1), |state| {
+                println!(
+                    "[Upgrade at height {}] Deploy simple contract",
+                    state.block_height()
+                );
+
+                // create a message for deploying the contract
+                let mut mf =
+                    SignedMessageFactory::new_secp256k1(my_secret_key(), 1, state.chain_id());
+                let message = match mf
+                    .fevm_create(
+                        Bytes::from(
+                            hex::decode(CONTRACT_HEX)
+                                .context("error parsing contract")
+                                .unwrap(),
+                        ),
+                        Bytes::default(),
+                        TokenAmount::default(),
+                        GAS_PARAMS.clone(),
+                    )
+                    .unwrap()
+                {
+                    ChainMessage::Signed(signed) => signed.into_message(),
+                    _ => panic!("unexpected message type"),
+                };
+
+                // execute the message
+                let (res, _) = state.execute_implicit(message).unwrap();
+                assert!(
+                    res.msg_receipt.exit_code.is_success(),
+                    "{:?}",
+                    res.failure_info
+                );
+
+                // parse the message receipt data and make sure the contract was deployed to the expected address
+                let res = fvm_ipld_encoding::from_slice::<eam::CreateReturn>(
+                    &res.msg_receipt.return_data,
+                )
+                .unwrap();
+                assert_eq!(
+                    res.delegated_address(),
+                    Address::from_str(CONTRACT_ADDRESS).unwrap()
+                );
+
+                Ok(())
+            })
+            .unwrap(),
+        )
+        .unwrap();
+
+    upgrade_scheduler
+        .add(
+            Upgrade::new("mychain", 2, None, |state| {
+                println!(
+                    "[Upgrade at height {}] Sends a balance",
+                    state.block_height()
+                );
+
+                // build the calldata for the send_coin function
+                let (client, _mock) = ethers::providers::Provider::mocked();
+                let simple_coin = SimpleCoin::new(EthAddress::from_id(101), client.into());
+                let call = simple_coin.send_coin(
+                    // the address we are sending the balance to (which is us in this case)
+                    EthAddress::from(my_secret_key().public_key()).into(),
+                    // the amount we are sending
+                    U256::from(SEND_BALANCE_AMOUNT),
+                );
+
+                // create a message for sending the balance
+                let mut mf =
+                    SignedMessageFactory::new_secp256k1(my_secret_key(), 1, state.chain_id());
```

Comment:
See you are using the same nonce and same secret key here, which means if this migration also tried to deploy a contract, it would compute the same address.